### ISSUE TITLE
Using `json` field hangs with high CPU (with Deno)

### ISSUE THREAD

**Issue by raidokaldma** [sindresorhus/ky#341]
I'm using latest version of Deno (version 1.9.2) and trying out an example from https://github.com/sindresorhus/ky#usage

Using `json: {...}` in options does not yield expected result. Process hangs with CPU at 100%.

### CPU @ 100%, process never finishes
```typescript
// deno run --allow-net will_never_finish.ts
import ky from "https://cdn.skypack.dev/ky?dts";
const json = await ky.post("https://example.com", { json: { foo: true } })
  .text();
console.log(json);
```

### Using `body: JSON.stringify()` instead of `json:` works
```typescript
// deno run --allow-net will_finish.ts
import ky from "https://cdn.skypack.dev/ky?dts";
const json = await ky.post("https://example.com", {
  headers: { "content-type": "application/json" },
  body: JSON.stringify({ foo: true }),
}).text();
console.log(json);
```

**Comment by raidokaldma** [comment]
Actually the example with `json` eventually runs out of memory:
```
$ deno run --allow-net will_never_finish.ts

<--- Last few GCs --->

[919969:0x2b7c00000000]    54749 ms: Scavenge 1397.8 (1439.3) -> 1397.4 (1439.3) MB, 0.9 / 0.0 ms  (average mu = 0.150, current mu = 0.109) allocation failure 
[919969:0x2b7c00000000]    54751 ms: Scavenge 1397.8 (1439.3) -> 1397.4 (1439.3) MB, 0.8 / 0.0 ms  (average mu = 0.150, current mu = 0.109) allocation failure 
[919969:0x2b7c00000000]    54754 ms: Scavenge 1398.1 (1439.5) -> 1397.6 (1439.5) MB, 0.8 / 0.0 ms  (average mu = 0.150, current mu = 0.109) allocation failure 


<--- JS stacktrace --->


#
# Fatal javascript OOM in Ineffective mark-compacts near heap limit
#

[1]    919969 trace trap (core dumped)  deno run --allow-net will_never_finish.ts
```

**Comment by Drache93** [comment]
Looks like a Deno issue. Appears in 1.9.1>=. 1.9.0 works fine.

**Comment by sholladay** [comment]
Thanks for all the details, that's very helpful. It's odd that you had better luck with plain `JSON.stringify()`, because we're not doing much more than that under the hood, at least in terms of special casing the `json` option. My guess is that the bug has to do with Deno's implementation of `request.clone()`, which we use in a few places regardless of whether you're using `json` mode or not. I've had all kinds of problems with `clone()` in the past, on various platforms, most of which have to do with how the body is copied to the new request.

**Comment by sholladay** [comment]
@raidokaldma sure, but first do you have a link to a Deno issue or PR that you could link to in case someone else comes across this and wants to dig into it?

**Comment by raidokaldma** [comment]
deno 1.10.1 got released 4 hours ago. Newer version fixes the issue. I think there is no need to spend any more time on this issue. I'll close it.

**Comment by sholladay** [comment]
They routinely revert patches, so it's not ideal to assume it's fixed permanently and move on. It's best to leave a link to the relevant work in case we need to circle back on it.

From a quick reading of the recent [release notes](https://github.com/denoland/deno/releases), this PR stood out to me as likely being the one that fixed it: https://github.com/denoland/deno/pull/10406

That would imply that the minimum fixed version is Deno `1.10.0`, not `1.10.1`. But I haven't tested.

Anyway, glad this is resolved. :)

**Comment by raidokaldma** [comment]
That was some good investigation work, thank you for spending your time on it!
